src/main/java/org/terasology/gookeeper/system/EconomySystem.java

```diff
@@ -75,32 +93,120 @@
     @In
     private NUIManager nuiManager;
 
+    @In
+    private BlockCommands blockCommands;
+
+    @In
+    private ItemCommands itemCommands;
+
     private static final Logger logger = LoggerFactory.getLogger(EconomySystem.class);
     private Random random = new FastRandom();
     private static final float baseEntranceFee = 100f;
     private static final float baseVisitFee = 10f;
-    private static boolean setHud = false;
 
     @Override
     public void initialise() {
     }
 
+    @ReceiveEvent
+    public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, InventoryComponent inventory) {
+        if (player.hasComponent(EconomyComponent.class)) {
+            nuiManager.getHUD().addHUDElement("PlayerHud");
+        }
+    }
+
     @Override
     public void shutdown() {
     }
 
     @Override
     public void update(float delta) {
```

Comment:
If update is never used then we should be able to drop the `UpdateSubscriberSystem`